### PR TITLE
Added requests session paramater to make it accept an existing session

### PR DESCRIPTION
**Pull request by p0psicles**
Will help when CacheControl is used for the session.

tl;dr
- renamed reserved keywords like type and id. It's won't break anything. But it's best practice to not use these as vars.
- Pylint cleanup
- Added the option to use the systems tmp cache dir.
- Raised exception when the <error>banned</error> message presents itself. To make sure it's handled when called externally.
- Some .format changes {} to {0}
- Updated the requirements.txt for the additional imports

Reason of the PR, is that i'm looking into adding anidb as an indexer for medusa. Although a recently noticed there are allot of obstacles to overcome. And anidb isn't really cooperative. At least with this lib, we could also start using the http client for anidb's hot anime/popular anime lists.

### CODE REVIEW

**Comment by p0psicles** [comment]
Just noticed that if going with this PR, the following import isn't needed anymore: https://github.com/p0psicles/simpleanidb/blob/develop/simpleanidb/models.py#L2

File: simpleanidb/__init__.py [hunk 1]
```diff
@@ -54,7 +79,7 @@ def search(self, term):
                 if term in title.text.lower():
                     anime_ids.append((int(anime.get("aid")), anime))
                     break
-        return [Anime(self, aid, False, xml_node) for aid, xml_node in anime_ids]
+        return [Anime(self, aid, autoload, xml_node) for aid, xml_node in anime_ids]
```
Comment (p0psicles) [inline simpleanidb/__init__.py]: I'm starting the think why you've set this to false. As i was testing it, and when enabled you'l get banned in no time.
